src/MongoDB.Driver.Core/Core/Operations/RetryableWriteCommandOperationBase.cs

```diff
@@ -184,5 +204,66 @@ public Task<BsonDocument> ExecuteAttemptAsync(RetryableWriteContext context, int
         /// <param name="transactionNumber">The transaction number.</param>
         /// <returns>A command.</returns>
         protected abstract BsonDocument CreateCommand(ConnectionDescription connectionDescription, int attempt, long? transactionNumber);
+
+        /// <summary>
+        /// Creates the command payloads.
+        /// </summary>
+        /// <returns>The command payloads.</returns>
+        protected abstract IEnumerable<Type1CommandMessageSection> CreateCommandPayloads(IChannelHandle channel, int attempt);
+
+        // private methods
+        private MessageEncoderSettings CreateMessageEncoderSettings(IChannelHandle channel)
+        {
+            var clone = _messageEncoderSettings.Clone();
+            clone.Add(MessageEncoderSettingsName.MaxDocumentSize, channel.ConnectionDescription.MaxDocumentSize);
+            clone.Add(MessageEncoderSettingsName.MaxMessageSize, channel.ConnectionDescription.MaxMessageSize);
+            clone.Add(MessageEncoderSettingsName.MaxWireDocumentSize, channel.ConnectionDescription.MaxWireDocumentSize);
+            return clone;
```

Comment:
These values were not yet known at the time the encoder setting that were passed in was created.

Now that we have a channel we can add this additional data.

We clone the settings so as to not modify the settings passed in.